### PR TITLE
Create chopped_nginx cookbook to mange NGINX; use it to proxy factorio-api

### PR DESCRIPTION
**Pull request by justjake**
The new NGINX cookbook is my take on NGINX config management in Chef. Instead of sketchy templates, we use a Ruby DSL that is guaranteed to generate a syntactically-valid NGINX config files. Maybe I should spin config generation DSL into its own gem? Whatever. MONOREPO WOO!!!

Anyways now that we can configure NGINX, I've set up NGINX to reverse-proxy factorio-api using a Unix domain socket. It works. Factorio-api things are happening on port 80.

Why a pull request in my own repo? SELF CODE REVIEW BABYYYYYY. Time to find the things I forget to do and zap them before I merge

### CODE REVIEW

**Comment by philihp** [comment]
dear god what done indeed

**Comment by fwenzel** [comment]
That's quite the changeset!

And for the record PRs against your own repo is a perfectly acceptable pattern.

File: site-cookbooks/chopped_nginx/README.md [hunk 1]
```diff
@@ -10,6 +10,15 @@ Goals of this cookbook:
 - generate clean NGINX config files from ruby data structures
 - cover common NGINX use cases with sensible defaults
 
+## TODOs
+
+- refactor libraries/chopped_nginx.rb; it's too big as-is
```
Comment (justjake) [inline site-cookbooks/chopped_nginx/README.md]: DONE. Should remove this TODO

File: site-cookbooks/chopped_nginx/README.md [hunk 2]
```diff
@@ -10,6 +10,15 @@ Goals of this cookbook:
...
+- make some typical use case templates? model these off what we use at armada.systems
+- make sure that the chef run checks each resource as it is declared. right now
```
Comment (justjake) [inline site-cookbooks/chopped_nginx/README.md]: DONE. Horrible idea. Should remove

**Comment by justjake** [comment]
LGTM

**Pull request merged**